frappe/public/js/frappe/form/grid_row.js

```diff
@@ -1436,7 +1436,9 @@ export default class GridRow {
 		let field = this.on_grid_fields_dict[fieldname];
 		// reset field value
 		if (field) {
-			field.docname = this.doc.name;
+			// the below if statement is added to factor in the exception when this.doc is undefined -
+			// - after row removals via customize_form.js on links, actions and states child-tables
+			if (this.doc) field.docname = this.doc.name;
```

Comment:
Hi @akhilnarang, when a grid-row/child-table row (for links, actions, states) is deleted using a 'Customize Form' view, then the deleted grid-row document (this.doc) in locals (browser copy of the doctype) becomes undefined, because that particular row gets deleted - in this case when the refresh_field method in grid_row.js is called, the console throws a silent error mentioning that this.doc is undefined (in statement 1441 above), where field.docname is assigned the value from this.doc.name